plugins/hako/src/tasks/delete.ts

```diff
@@ -15,7 +16,16 @@ const HakoDeleteSchema = z
       .describe(
         'name of the app with the ephemeral app to delete (will be the same as the name of the docker image)'
       ),
-    ephemeralId: z.string().describe('ID that is used by Hako to identify a particular ephemeral app'),
+    asReviewApp: z
+      .boolean()
+      .default(false)
+      .describe(
+        'whether to delete a temporary review app. overrides the `ephemeralId` option with its own hash of the git branch.'
+      ),
+    ephemeralId: z
```

Comment:
thought: is there an opportunity to merge both options of `ephemeralId` and `asReviewApp` into just `ephemeralId`? For example, if an `ephemeralId` is not provided, can we default to the hash of the git branch?

_Context:_ Now that we have made `ephemeralId` optional, what happens if someone sets `asReviewApp` as `false` and also doesn't set an `ephemeralId`?

Reply:
I think I'd prefer something more consistent and explicit like
```ts
target: z.union([
  z.object({ productionApp: z.literal(true) }),
  z.object({ reviewApp: z.literal(true) }),
  z.object({ ephemeralApp: z.string() })
])
```
but yeah i don't think we need to make a change like this yet and the overlap between the `HakoDeploy` and `HakoDelete` truth tables mean that the current schema is fine.

Reply:
yeah, i can see it potentially being a footgun. as a failsafe we could throw an error if neither option is defined, and add another option that's like `yesReallyDeleteUnsuffixedAppTrustMe`?